pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -323,19 +332,34 @@ public MessageDupStatus isDuplicate(PublishContext publishContext, ByteBuf heade
 
         String producerName = publishContext.getProducerName();
         long sequenceId = publishContext.getSequenceId();
+        // The process of the Producer sending chunk messages is continuous, and all chunks of a message use the same
+        // message metadata and sequence ID. Therefore, it is only necessary to check if the sequence ID of the first
+        // chunk is duplicated.
+        // When we receive the initial message of a non-duplicated chunk message, we place it in the
+        // chunkMessageOngoing. Upon completion of sending this chunk message, if we receive other messages
+        // sent by this Producer, we will remove it from the chunkMessageOngoing.
+        headersAndPayload.markReaderIndex();
+        MessageMetadata msgMetadata = Commands.parseMessageMetadata(headersAndPayload);
+        headersAndPayload.resetReaderIndex();
+        if (chunkMessageOngoing.containsKey(producerName)) {
+            //Deduplication requires the dependency on the ordered nature of messages to work accurately.
+            if (publishContext.isChunked() && chunkMessageOngoing.get(producerName) < msgMetadata.getChunkId()) {
+                chunkMessageOngoing.put(producerName, msgMetadata.getChunkId());
+                return MessageDupStatus.NotDup;
+            } else {
+                chunkMessageOngoing.remove(producerName);
```

Comment:
Maybe optimize this if-else code a little to remove only when it contains?

Reply:
I thought about it again, and it seems that we can leave `chunkMessageOngoing` out. Could you take a look at this new plan?